server/odc-core/src/main/java/com/oceanbase/odc/core/sql/util/JdbcDataTypeUtil.java

```diff
@@ -201,6 +201,14 @@ public static Object getValueFromStatement(CallableStatement statement, int inde
 
     public static Object getValueFromResultSet(ResultSet resultSet, int index, String type)
             throws SQLException {
+        // YEAR type is used to represent the year in 4-digit format. The range is 1901 to 2155, and 0000
```

Comment:
I don't think add if before switch is a good solution.
YEAR is a special MySQL datatype, although it stored in integer, but is have some special usage.
eg year start begin 1900, and if you use in where clause, value less than 1900 is  special number need to be handled. So why not create it as a type and process it by type referenced function

Reply:
![image](https://github.com/user-attachments/assets/8f98e550-89e6-475c-bbc1-5328186a3c4a)
![image](https://github.com/user-attachments/assets/f48b0d2a-e5a3-4da6-be93-74b655ea45a0)
The enum type `JDBCType` belongs to the package `java.sql`  and cannot be modified. The `year` type is not included

Reply:
Using` getShort` will only get the values from 1901 to 2155 and 0

Reply:
For getting function execution results from database which dialect_type is mysql, use `JdbcRowMapper` to handle the type conversion